src/modules/dataFetching/components/GenericTableWithData.tsx

```diff
@@ -61,6 +61,7 @@ export interface Props<
   defaultFilterValues?: Partial<FilterOptionsType>;
   showTopToolbar?: boolean;
   noSort?: boolean;
+  noPaginate?: boolean;
```

Comment:
I added `noPaginate` for now just to get everything hooked up. But I would like to discuss what to do in terms of pagination, sorting, and filtering on the `FormProjectMatchTable`.

The underlying record type, `InstanceProjectMatch` is just a ruby object, not a rails model. My understanding is that from the backend's perspective, there's not much benefit to paginating this list, because we aren’t querying the db.

Is it worthwhile to enable pagination for this table just for the ux benefit? Is there anywhere else in the codebase already where we paginate, filter, or sort on a non-ActiveRecord?

Reply:
If it's not too hard, I think it would be worthwhile to resolve the array as paginated in the GraphQL API, so that the frontend can use the same patterns for pagination.

We have almost had this need before (paginating an array) and I'm sure we'll have it again. I got a partial (or full?) implementation one time that never made it into the codebase. I think it will involve something  like:
* `Types::Forms::ProjectMatch.page_type`
* Update [this function](https://github.com/greenriver/hmis-warehouse/blob/771099a1842631e424086ceadd617fec33ded06c/drivers/hmis/app/graphql/types/paginated_scope.rb#L18) to have conditional logic for how to apply `limit` or `offset` based on whether `@all_nodes` is an AR scope or an array. 
  * Actually, maybe the more appropriate thing is to create a `PaginatedArray` that is similar to `PaginatedScope`, and choose the appropriate one [here](https://github.com/greenriver/hmis-warehouse/blob/771099a1842631e424086ceadd617fec33ded06c/drivers/hmis/app/graphql/types/base_field.rb#L58-L59) based on whether the object is an array or a scope.
* There is a downside which is we could accidentally resolve an array when really we should use a scope, and it would be slow. For that reason, it might be better to have something explicit (eg specifying `Types::Forms::ProjectMatch.array_page_type` but that might be a little tricker to work in

Reply:
Yes sounds good! See https://github.com/greenriver/hmis-warehouse/pull/4494/commits/a67d6179b9971b5f6cb635c81378734437183380